Spigot-Server-Patches/0722-More-Enchantment-API.patch

```diff
@@ -0,0 +1,184 @@
++            return EnumMonsterType.UNDEFINED;
++        } else if (entityCategory == EntityCategory.UNDEAD) {
++            return EnumMonsterType.UNDEAD;
++        } else if (entityCategory == EntityCategory.ARTHROPOD) {
++            return EnumMonsterType.ARTHROPOD;
++        } else if (entityCategory == EntityCategory.ILLAGER) {
++            return EnumMonsterType.ILLAGER;
++        } else if (entityCategory == EntityCategory.WATER) {
++            return EnumMonsterType.WATER_MOB;
++        }
++
++        throw new IllegalArgumentException(String.format("%s is an unrecognized entity category", entityCategory));
++    }
+     // Paper end
+ }
+diff --git a/src/test/java/io/papermc/paper/enchantments/EnchantmentRarityTest.java b/src/test/java/io/papermc/paper/enchantments/EnchantmentRarityTest.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..cf4093bd62b244eb8f3e376ff4c94c1e4ba2f865
+--- /dev/null
++++ b/src/test/java/io/papermc/paper/enchantments/EnchantmentRarityTest.java
+@@ -0,0 +1,17 @@
++package io.papermc.paper.enchantments;
++
++import net.minecraft.world.item.enchantment.Enchantment.Rarity;
++import org.bukkit.craftbukkit.enchantments.CraftEnchantment;
++import org.junit.Test;
++
++import static org.junit.Assert.assertNotNull;
++
++public class EnchantmentRarityTest {
++
++    @Test
++    public void test() {
++        for (Rarity nmsRarity : Rarity.values()) {
++            CraftEnchantment.fromNMSRarity(nmsRarity);
++        }
++    }
++}
+diff --git a/src/test/java/io/papermc/paper/entity/EntityCategoryTest.java b/src/test/java/io/papermc/paper/entity/EntityCategoryTest.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..eef9866858d86b377035289395e831d5925bdfbe
+--- /dev/null
++++ b/src/test/java/io/papermc/paper/entity/EntityCategoryTest.java
+@@ -0,0 +1,31 @@
++package io.papermc.paper.entity;
++
++import com.google.common.collect.Sets;
++import net.minecraft.world.entity.EnumMonsterType;
++import org.bukkit.craftbukkit.entity.CraftLivingEntity;
++import org.bukkit.entity.EntityCategory;
++import org.junit.Test;
++
++import java.lang.reflect.Field;
++import java.util.Set;
++
++import static org.junit.Assert.assertTrue;
++
++public class EntityCategoryTest {
++
++    @Test
++    public void test() throws IllegalAccessException {
++
++        Set<EnumMonsterType> enumMonsterTypeSet = Sets.newHashSet();
++        for (Field field : EnumMonsterType.class.getDeclaredFields()) {
++            if (field.getType() == EnumMonsterType.class) {
++                enumMonsterTypeSet.add((EnumMonsterType) field.get(null));
++            }
++        }
++
++        for (EntityCategory entityCategory : EntityCategory.values()) {
++            enumMonsterTypeSet.remove(CraftLivingEntity.fromBukkitEntityCategory(entityCategory));
++        }
++        assertTrue(String.format("Found a %s instance that is missing a bukkit counterpart", EnumMonsterType.class), enumMonsterTypeSet.size() == 0);
```

Comment:
Print them

Reply:
Print the ones left over? Is that even useful? It’s just gonna be EnumMonsterClass@276ae76 or smth

Reply:
Ok, I just changed it to a Map with the field name and printed that.